### ISSUE TITLE
Avg files over different source groups.

### ISSUE THREAD

**Issue by ckoerber** [callat-qcd/nucleon_elastic_FF#19]
Provide API to average different files over source groups.

- The file names will differ by their source group label `{base_name}_srcs{n1}-{n2}.h5`
- The internal data set names will not change
- The user should be able to provide a list of input files matching the above pattern with different `n1` and `n2`
- The order of input files does not matter
- The user should be able to provide an output file name

**Comment by ckoerber** [comment]
With the above commit, I have pushed the `h5avg` command line script to the repo ([`concat` branch](https://github.com/callat-qcd/nucleon_elastic_FF/tree/concat)). After `pip` reinstalling the scripts, you should be able to run:

```
> h5avg -h
usage: h5avg [-h] [--input INPUT [INPUT ...]] [--output OUTPUT] [--overwrite]

Reads h5 files and exports the average of datasets across files. Each group in
the file list will be averaged over files. Fails if it finds less dsets then
specified files for each group.

optional arguments:
  -h, --help            show this help message and exit
  --input INPUT [INPUT ...], -i INPUT [INPUT ...]
                        Files to average (list). Must be given.
  --output OUTPUT, -o OUTPUT
                        Name of the output file. Must be given.
  --overwrite, -f       Overwrite hdf5 files if they already exist.
                        [default=False]
```

@walkloud: Let me know if this is what you have asked for.

This routine wraps the existing [`nucleon_elastic_ff.data.scripts.average.dset_avg`](https://github.com/callat-qcd/nucleon_elastic_FF/blob/concat/nucleon_elastic_ff/data/scripts/average.py) method, so it is also indirectly unittested with our legacy tests.

**Comment by ckoerber** [comment]
Closed with PR #21
